src/pages/data-source/[id].vue

```diff
@@ -313,22 +318,35 @@ function getPrev() {
   return searchStore.mostRecentSearchIds[previousIdIndex.value];
 }
 
-onMounted(() => {
-  handleShowMoreButton();
-  window.addEventListener('resize', handleShowMoreButton);
-});
+let observer = null;
```

Comment:
Since we're instantiating a new `ResizeObserver` on every change of `descriptionRef`, I'd suggest storing this in a ref object...

```suggestion
const observer = ref(null);
```